scanpy/external/pp/_scrublet.py

```diff
@@ -0,0 +1,511 @@
+        'expected_doublet_rate': expected_doublet_rate,
+        'sim_doublet_ratio': adata_sim.uns['scrublet']['parameters'][
+            'sim_doublet_ratio'
+        ],
+        'n_neighbors': n_neighbors,
+        'random_state': random_state,
+    }
+
+    if get_doublet_neighbor_parents:
+        adata_obs.uns['scrublet'][
+            'doublet_neighbor_parents'
+        ] = scrub.doublet_neighbor_parents_
+
+    return adata_obs
+
+
+def simulate_doublets(
+    adata: AnnData,
+    raw_layer: str = 'raw',
+    sim_doublet_ratio: float = 2.0,
+    synthetic_doublet_umi_subsampling: float = 1.0,
+    random_seed: int = 0,
+) -> AnnData:
+
+    """\
+    Simulate doublets by adding the counts of random observed transcriptome pairs.
+
+    Arguments
+    ---------
+    adata
+        The annotated data matrix of shape ``n_obs`` × ``n_vars``. Rows
+        correspond to cells and columns to genes. Genes should have been
+        filtered for expression and variability, and the object should contain
+        raw expression of the same dimensions. 
+        
+    raw_layer
+        Layer of adata where raw values are stored, or 'X' if values are in .X. 
+    
+    sim_doublet_ratio
+        Number of doublets to simulate relative to the number of observed 
+        transcriptomes. If `None`, self.sim_doublet_ratio is used.
+
+    synthetic_doublet_umi_subsampling
+        Rate for sampling UMIs when creating synthetic doublets. If 1.0, 
+        each doublet is created by simply adding the UMIs from two randomly 
+        sampled observed transcriptomes. For values less than 1, the 
+        UMI counts are added and then randomly sampled at the specified
+        rate.
+
+    Returns
+    -------
+    adata : anndata.AnnData with simulated doublets in .X
+        if ``copy=True`` it returns or else adds fields to ``adata``:
+            ``adata.uns['scrublet']['doublet_parents']`` 
+                Pairs of ``.obs_names`` used to generate each simulated doublet transcriptome
+            ``uns['scrublet']['parameters']``
+                Dictionary of Scrublet parameters
+    """
+    try:
+        import scrublet as sl
+    except ImportError:
+        raise ImportError(
+            'Please install scrublet: `pip install scrublet` or `conda install scrublet`.'
+        )
+
+    if raw_layer == 'X':
+        scrub = sl.Scrublet(adata.X)
+    else:
+        scrub = sl.Scrublet(adata.layers[raw_layer])
+
+    scrub.simulate_doublets(
+        sim_doublet_ratio=sim_doublet_ratio,
+        synthetic_doublet_umi_subsampling=synthetic_doublet_umi_subsampling,
+    )
+
+    adata_sim = AnnData(scrub._E_sim)
+    adata_sim.obs['n_counts'] = scrub._total_counts_sim
+    adata_sim.obsm['doublet_parents'] = scrub.doublet_parents_
+    adata_sim.uns['scrublet'] = {}
+    adata_sim.uns['scrublet']['parameters'] = {}
+    adata_sim.uns['scrublet']['parameters']['sim_doublet_ratio'] = sim_doublet_ratio
+    return adata_sim
+
+
+def plot_histogram(
```

Comment:
If this is going to be exported from a high level, could it get a name that ties it to scrublet? Maybe `sce.pl.scrublet_score_distribution`?

Reply:
Now moved to sce.pl and named as suggested.